plutus-tx-plugin/test/Plugin/Functions/recursive/even.pir.golden

```diff
@@ -45,7 +51,20 @@
       (vardecl
         subtractInteger (fun (con integer) (fun (con integer) (con integer)))
       )
-      (lam x (con integer) (lam y (con integer) [ [ subtractInteger x ] y ]))
+      (lam
+        x
+        (con integer)
+        (lam
+          y
+          (con integer)
+          (let
+            (nonrec)
+            (termbind (strict) (vardecl x (con integer)) x)
```

Comment:
Right, I think I've worked out where these come from. A bang pattern on a function argument basically becomes:
```
f !x = t
--->
f x = case x2@x of { _ -> t[x2/x] }
```
The case gets dropped but we keep the strict let-binding from the @ pattern, which creates these things for... all arguments basically.